mkosi/config.py

```diff
@@ -1634,11 +1633,13 @@ class UKIProfile:
     cmdline: list[str]
 
 
-def make_simple_config_parser(settings: Sequence[ConfigSetting], type: type[Any]) -> Callable[[str], Any]:
+def make_simple_config_parser(
+    settings: Sequence[ConfigSetting[Any]], valtype: type[T]
```

Comment:
I think `object` is better than `Any` here since it doesn't disable type checking, ditto elsewhere